### PR TITLE
feat: add unix passwod reset to security web ui

### PR DESCRIPTION
**Pull request by theSuess**
Adds the functionality to change your own unix password. Currently, the button is displayed for all users as I did not want to copy the user fetching code from `profile.rs`. If anyone has an idea on how to solve this in a clean manner, I'd be happy to implement it.

- [x] cargo fmt has been run
- [x] cargo clippy has been run (only existing issues showed up)
- [x] cargo test has been run and passes

### CODE REVIEW

**Comment by theSuess** [comment]
Forgot to clear the input on success/failure.

Here's a short screenrecord on how it looks like

[pw_change_flow.webm](https://user-images.githubusercontent.com/1364576/188279288-2f3e4202-2ef0-4cec-b260-3a4c83f0d4e7.webm)

File: kanidmd_web_ui/src/views/security.rs
Comment (yaleman) [inline kanidmd_web_ui/src/views/security.rs:59]: This feels like it should be an `Option<String>` just so we're never passing *any* string around unless it's actually in use.

File: kanidmd_web_ui/src/views/security.rs
Comment (yaleman) [inline kanidmd_web_ui/src/views/security.rs:153]: Why not just use an assignment here? That feels weird.

**Review by yaleman — CHANGES_REQUESTED** [review]
Thanks for submitting this so quickly 😄 

This definitely needs the usual "password" + "confirm password" boxes; even though it's trivial to update when the user goes "oh no I can't log in, let me fix it" it's one of those UI patterns that saves some headaches for everyone.

Regarding the "shows for everyone" thing, that's... something that needs pondering. There's a bunch of helper functions for this kind of introspection that need writing at the moment... 🤔

File: kanidmd_web_ui/src/views/security.rs
Comment (theSuess) [reply kanidmd_web_ui/src/views/security.rs:153]: I've taken this from the input field on the login page. I'm pretty new to this framework so I assumed it has a reason. Will have to change this anyways when switching to options.

File: kanidmd_web_ui/src/views/security.rs
Comment (theSuess) [reply kanidmd_web_ui/src/views/security.rs]: This became obsolete due to refactoring

File: kanidmd_web_ui/src/views/security.rs
Comment (theSuess) [reply kanidmd_web_ui/src/views/security.rs]: Also obsolete now

**Comment by theSuess** [comment]
I've refactored the `ViewsApp` to be the owner of the profile data (this also allows us to display the username in the header, etc.).
Now the change password dialog is displayed conditionally

Currently, this is all realized using `props` but ideally we should introduce a global `Agent` (yew term for `Service` or `Store`) to take care of global state/api requests. But this could be the topic of a separate PR

**Comment by Firstyear** [comment]
Well for now I'm happy for it to "display for everyone", but @yaleman is right, we'll need to refactor once we can integrate some better permissions of "what to display" into the user auth token.

File: kanidm_proto/src/v1.rs
Comment (Firstyear) [inline kanidm_proto/src/v1.rs:327]: Why does this need eq?

File: kanidmd_web_ui/src/components/change_unix_password.rs
Comment (Firstyear) [inline kanidmd_web_ui/src/components/change_unix_password.rs:136]: I think it would be good here to have the title to match saying "Update your unix password" along with some short text about what this password is used for. Something like "This password is used when logging into a unix-like system"

File: kanidmd_web_ui/src/views/security.rs
Comment (Firstyear) [inline kanidmd_web_ui/src/views/security.rs:191]: 😅 Thanks for fixing this, I was in the middle of a large account refactor.

File: kanidmd_web_ui/src/components/change_unix_password.rs
Comment (Firstyear) [inline kanidmd_web_ui/src/components/change_unix_password.rs:69]: Totally up to you if you want to do this, but there is a way to prevent submit being possible until both fields match, I think it's in the credential update code. If you want you can update to use a similar pattern here which saves the need to error on the fields not matching.

File: kanidmd_web_ui/src/components/change_unix_password.rs
Comment (Firstyear) [inline kanidmd_web_ui/src/components/change_unix_password.rs:80]: Worth checking the password update code in the other credential update path since I think this can return a password quality errer from zxcvbn, and maybe there is a way to share some code between the two?

**Review by Firstyear — CHANGES_REQUESTED** [review]
Looks really good some minor comments and questions, and some suggestions for possible improvements, but I'd also be happy to merge and add the other improvements in a second PR too :) 

Please also add yourself to the CONTRIBUTORS.md file since you are contributing!

File: kanidm_proto/src/v1.rs
Comment (theSuess) [reply kanidm_proto/src/v1.rs:327]: We need this since we pass the `WhoamiResponse` as a property now (to enable conditional display of the dialog)

I'm not too sure of the implications but if you'd like this to stay the same, we can always add a new struct for the property (which will become obsolete once better state management is implemented in the UI)

**Comment by theSuess** [comment]
> Well for now I'm happy for it to "display for everyone", but @yaleman is right, we'll need to refactor once we can integrate some better permissions of "what to display" into the user auth token.

With the latest changes, the button is displayed conditionally (see https://github.com/kanidm/kanidm/pull/1014#issuecomment-1236301628 for how it's implemented now and how it could be implemented in a better way)

File: kanidmd_web_ui/src/components/change_unix_password.rs
Comment (theSuess) [reply kanidmd_web_ui/src/components/change_unix_password.rs:85]: Resolved. Also added a little message to the password change modal in the credentials update why the field is marked as invalid

File: kanidmd_web_ui/src/components/change_unix_password.rs
Comment (theSuess) [reply kanidmd_web_ui/src/components/change_unix_password.rs:96]: While implementing the blocking of the submit button, I figured out just how similar the two modals are. I'll probably work on a shared component after this is merged.

**Comment by theSuess** [comment]
After my last changes, I figured out that I just reimplemented the existing modal, so I'll work on merging these two after this PR is merged

File: kanidm_proto/src/v1.rs
Comment (Firstyear) [reply kanidm_proto/src/v1.rs:327]: I think it's okay, I was just curious.

**Comment by Firstyear** [comment]
I think we just need you to rebase to latest master and then we can merge :)

**Comment by theSuess** [comment]
The branch is rebased now. Should I also commit the built frontend (in pkg) or is there some kind of CI in place for that?

**Comment by yaleman** [comment]
Please ensure the frontend WASM’s been built and committed :)

**Comment by theSuess** [comment]
Done! I just ran the `./build_wasm.sh` script. Hope this is correct

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
@yaleman Can you double check, your request change is blocking merge :)

**Review by yaleman — APPROVED** [review]

**Comment by yaleman** [comment]
yolooooo

**Pull request merged**
